routing/index_graph.cpp

```diff
@@ -1,9 +1,94 @@
+    // restrictricted U-turn it's restricted to make a U-turn on the both ends of the feature.
+    // Generally speaking it's wrong. In osm there's information about the end of the feature
+    // where the U-turn is restricted. It's necessary to pass the data to mwm and to use it here.
+    // Please see test LineGraph_RestrictionF1F1No for details.
+    return u.GetSegmentIdx() == v.GetSegmentIdx() && u.IsForward() != v.IsForward();
+  }
+
+  // Taking into account that number of restrictions of type Only starting and
+  // ending with the same feature id is relevantly small it's possible to ignore such cases.
+  if (featureIdFrom == featureIdTo)
+    return false;
+
+  // Looking for a range of restrictins of type Only starting from |featureIdFrom|
+  // and finishing with any feature.
+  auto const range = equal_range(restrictions.cbegin(), restrictions.cend(),
+                                 Restriction(Restriction::Type::Only, {featureIdFrom}),
+                                 [](Restriction const & r1, Restriction const & r2) {
+                                   CHECK(!r1.m_featureIds.empty(), ());
+                                   CHECK(!r2.m_featureIds.empty(), ());
+                                   if (r1.m_type != r2.m_type)
+                                     return r1.m_type < r2.m_type;
+                                   return r1.m_featureIds[0] < r2.m_featureIds[0];
+                                 });
+  auto const lower = range.first;
+  auto const upper = range.second;
+
+  // Checking if there's at least one restrictions of type Only from |featureIdFrom| and
+  // ending with |featureIdTo|. If yes, returns false.
+  if (lower != upper && lower->m_featureIds[1] == featureIdTo)
+    return false;
+
+  // Checking if there's at least one Only starting from |featureIdFrom|.
+  // If not, returns false.
+  if (lower == restrictions.cend() || lower->m_type != Restriction::Type::Only ||
+      lower->m_featureIds[0] != featureIdFrom)
+  {
+    return false;
+  }
+
+  // Note. At this point it's clear that there is an item in |restrictions| with type Only
+  // and starting with |featureIdFrom|. So there are two possibilities:
+  // * |edges| contains |it->m_featureIds[1]| => the end of |featureIdFrom| which implies in
+  //   |restrictions| is considered.
+  // * |edges| does not contain |it->m_featureIds[1]| => either the other end or an intermediate
+  // point of |featureIdFrom| is considered.
+  // See test FGraph_RestrictionF0F2Only for details.
+  CHECK_EQUAL(lower->m_featureIds.size(), 2, ("Only two link restrictions are support."));
+  for (SegmentEdge const & e : edges)
+  {
+    if (e.GetTarget().GetFeatureId() == lower->m_featureIds[isOutgoing ? 1 /* to */ : 0 /* from */])
+      return true;
+  }
+  return false;
+}
+
+bool IsUTurn(Segment const & u, Segment const & v)
+{
+  return u.GetFeatureId() == v.GetFeatureId()
+      && u.GetSegmentIdx() == v.GetSegmentIdx()
```

Comment:
It's a old code. IsUTurn now looks like:
```
bool IsUTurn(Segment const & u, Segment const & v)
{
  return u.GetFeatureId() == v.GetFeatureId() && u.GetSegmentIdx() == v.GetSegmentIdx() &&
         u.IsForward() != v.IsForward();
}
```
Anyway I'll do clang format for all PR after review fixes.